core/trino-main/src/main/java/io/trino/type/UuidOperators.java

```diff
@@ -28,18 +33,62 @@
 import static io.trino.spi.function.OperatorType.CAST;
 import static io.trino.spi.type.UuidType.javaUuidToTrinoUuid;
 import static io.trino.spi.type.UuidType.trinoUuidToJavaUuid;
-import static java.util.UUID.randomUUID;
 
 public final class UuidOperators
 {
+    // All generators are thread safe
+    private static final RandomBasedGenerator V4_GENERATOR = Generators.randomBasedGenerator();
+    private static final TimeBasedReorderedGenerator V6_GENERATOR = Generators.timeBasedReorderedGenerator();
+    private static final TimeBasedEpochRandomGenerator V7_GENERATOR = Generators.timeBasedEpochRandomGenerator();
+
     private UuidOperators() {}
 
-    @Description("Generates a random UUID")
-    @ScalarFunction(deterministic = false)
+    @Description("Generates a random UUID v4 (RFC 4122)")
+    @ScalarFunction(deterministic = false, alias = "uuid_v4")
     @SqlType(StandardTypes.UUID)
     public static Slice uuid()
     {
-        java.util.UUID uuid = randomUUID();
+        java.util.UUID uuid = V4_GENERATOR.generate();
+        return javaUuidToTrinoUuid(uuid);
+    }
+
+    @Description("Generates a random UUID v6 (RFC-9562)")
+    @ScalarFunction(deterministic = false)
+    @SqlType(StandardTypes.UUID)
+    public static Slice uuid_v6()
+    {
+        java.util.UUID uuid = V6_GENERATOR.generate();
+        return javaUuidToTrinoUuid(uuid);
+    }
+
+    @Description("Generates a random UUID v6 from a given timestamp (RFC-9562)")
+    @ScalarFunction
+    @SqlType(StandardTypes.UUID)
+    @LiteralParameters("u")
+    @Constraint(variable = "u", expression = "min(u, 6)")
+    public static Slice uuid_v6(@SqlType("timestamp(u) with time zone") long rawTimestamp)
```

Comment:
That results in a rather nasty exception
```
trino> select uuid_v6( current_timestamp(9) );
Query 20251128_140821_00027_837vf failed: variable 'u' is already set to 9 when trying to set 6
java.lang.IllegalStateException: variable 'u' is already set to 9 when trying to set 6
```

Engine is not supposed to throw anything other than TrinoExceptions.